analysis/import/SourceCodeParser/src/main/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/oop/domain/metrics/OopMetricCalculationStrategy.kt

```diff
@@ -0,0 +1,28 @@
+package de.maibornwolff.codecharta.importer.sourcecodeparser.oop.domain.metrics
+
+import de.maibornwolff.codecharta.importer.sourcecodeparser.core.domain.metrics.MetricCalculationStrategy
+import de.maibornwolff.codecharta.importer.sourcecodeparser.core.domain.metrics.MetricMap
+import de.maibornwolff.codecharta.importer.sourcecodeparser.core.domain.metrics.MetricType
+import de.maibornwolff.codecharta.importer.sourcecodeparser.core.domain.tagging.Line
+import de.maibornwolff.codecharta.importer.sourcecodeparser.oop.domain.tagging.BranchTags
+import de.maibornwolff.codecharta.importer.sourcecodeparser.oop.domain.tagging.CodeTags
+import de.maibornwolff.codecharta.importer.sourcecodeparser.oop.domain.tagging.MethodTags
+
+class OopMetricCalculationStrategy: MetricCalculationStrategy {
+
+    override fun calculateMetrics(line: Line, previousMetrics: MetricMap): MetricMap {
+        return MetricMap(MetricType.values().map { it to singleMetric(it, line, previousMetrics) }.toMap())
+    }
+
+    private fun singleMetric(metricType: MetricType, line: Line, previousMetrics: MetricMap): Int{
+        return when(metricType){
+            MetricType.LoC -> line.lineNumber
```

Comment:
The calculation strategies for metrics (e.g loc or rloc) could be extracted into classes or strategies. I think your abstraction of source code allows reusing these strategies for other programming paradigmas than OOP. More complex metrics would make this when operation unreadable. This is just a note for the future. In my opinion this solution is good enough for this PR.